cms/search/tests/test_api_views.py

```diff
@@ -0,0 +1,138 @@
+        """Endpoint should return 404 when IS_EXTERNAL_ENV=False."""
+        response = self.client.get(RESOURCE_ENDPOINT)
+        self.assertEqual(response.status_code, 404)
+
+
+@override_settings(IS_EXTERNAL_ENV=False)
+class ResourceListViewPaginationTests(TestCase, ExternalAPITestMixin):
+    @classmethod
+    def setUpTestData(cls):
+        cls.index_page = IndexPageFactory(slug="custom-slug-1")
+        # One parent + 12 child pages = 13 total
+        cls.pages = InformationPageFactory.create_batch(
+            12, parent=cls.index_page, slug=factory.Sequence(lambda n: f"test_page_{n + 1}")
+        )
+        cls.total_resources = len(cls.pages) + 1  # +1 for the parent
+
+    def test_default_pagination_returns_first_slice(self):
+        """With no limit/ offset specified we should get DEFAULT_LIMIT (or all if fewer)."""
+        response = self.call_view_as_external("/v1/resources/")
+        self.assertEqual(response.status_code, 200)
+
+        data = self.parse_json(response)
+
+        expected_items = min(
+            self.total_resources, settings.SEARCH_API_DEFAULT_PAGE_SIZE
+        )  # Default page size is 20 currently
+        self.assertEqual(len(data["items"]), expected_items)
+        self.assertEqual(data["count"], expected_items)
+        self.assertEqual(data["total_count"], self.total_resources)
+        self.assertEqual(data["offset"], 0)
+
+    def test_second_slice_returns_remaining_items(self):
+        """Requesting offset=<7> should return whatever is left after the first slice."""
+        response = self.call_view_as_external(f"/v1/resources/?offset={7}")
+        self.assertEqual(response.status_code, 200)
+
+        data = self.parse_json(response)
+        expected_remaining = self.total_resources - 7
+        self.assertEqual(len(data["items"]), expected_remaining)
+        self.assertEqual(data["count"], expected_remaining)
+        self.assertEqual(data["offset"], 7)
+
+    def test_custom_limit_returns_requested_number(self):
+        response = self.call_view_as_external("/v1/resources/?limit=5")
+        self.assertEqual(response.status_code, 200)
+
+        data = self.parse_json(response)
+        self.assertEqual(len(data["items"]), 5)
+        self.assertEqual(data["count"], 5)
+        self.assertEqual(data["limit"], 5)
+
+    @patch("cms.search.pagination.CustomLimitOffsetPagination.max_limit", 10)
+    def test_limit_exceeds_max_uses_max(self):
+        """When limit exceeds max_limit, results should be capped at max_limit.
+        We add enough extra pages to go past MAX_LIMIT of 20.
+        """
+        response = self.call_view_as_external(f"/v1/resources/?limit={30}")
+        self.assertEqual(response.status_code, 200)
+
+        data = self.parse_json(response)
+        self.assertEqual(len(data["items"]), 10)
+        self.assertEqual(data["count"], 10)
+        self.assertEqual(data["limit"], 10)
+        self.assertEqual(data["total_count"], 13)
```

Comment:
nitpick: 
```suggestion
        self.assertEqual(data["total_count"], self.total_resources)
```

other than that LGTM

Reply:
Done!